pkg/operator/staticpod/controller/installerstate/installer_state_controller.go

```diff
@@ -0,0 +1,249 @@
+		LabelSelector: labels.SelectorFromSet(labels.Set{"app": "installer"}).String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	// collect all startingObjects that are in pending state for longer than maxToleratedPodPendingDuration
+	pendingPods := []*v1.Pod{}
+	for _, pod := range pods.Items {
+		if pod.Status.Phase != v1.PodPending || pod.Status.StartTime == nil {
+			continue
+		}
+		if c.timeNowFn().Sub(pod.Status.StartTime.Time) >= maxToleratedPodPendingDuration {
+			pendingPods = append(pendingPods, pod.DeepCopy())
+		}
+	}
+
+	// in theory, there should never be two installer startingObjects pending as we don't roll new installer pod
+	// until the previous/existing pod has finished its job.
+	foundConditions := []operatorv1.OperatorCondition{}
+	foundConditions = append(foundConditions, c.handlePendingInstallerPods(pendingPods)...)
+
+	// handle networking conditions that are based on events
+	networkConditions, err := c.handlePendingInstallerPodsNetworkEvents(pendingPods)
+	if err != nil {
+		return err
+	}
+	foundConditions = append(foundConditions, networkConditions...)
+
+	updateConditionFuncs := []v1helpers.UpdateStaticPodStatusFunc{}
+
+	// check the supported degraded foundConditions and check if any pending pod matching them.
+	for _, degradedConditionName := range degradedConditionNames {
+		// clean up existing foundConditions
+		updatedCondition := operatorv1.OperatorCondition{
+			Type:   degradedConditionName,
+			Status: operatorv1.ConditionFalse,
+		}
+		if condition := v1helpers.FindOperatorCondition(foundConditions, degradedConditionName); condition != nil {
+			updatedCondition = *condition
+		}
+		updateConditionFuncs = append(updateConditionFuncs, v1helpers.UpdateStaticPodConditionFn(updatedCondition))
+	}
+
+	if _, _, err := v1helpers.UpdateStaticPodStatus(c.operatorClient, updateConditionFuncs...); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *InstallerStateController) handlePendingInstallerPodsNetworkEvents(pods []*v1.Pod) ([]operatorv1.OperatorCondition, error) {
+	conditions := []operatorv1.OperatorCondition{}
+	if len(pods) == 0 {
+		return conditions, nil
+	}
+	namespaceEvents, err := c.eventsGetter.Events(c.targetNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, event := range namespaceEvents.Items {
+		if event.InvolvedObject.Kind != "Pod" {
+			continue
+		}
+		if !strings.Contains(event.Message, "failed to create pod network") {
+			continue
+		}
+		for _, pod := range pods {
+			if pod.Name != event.InvolvedObject.Name {
+				continue
+			}
+			condition := operatorv1.OperatorCondition{
+				Type:    "InstallerPodNetworkingDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  event.Reason,
+				Message: fmt.Sprintf("Pod %q on node %q observed degraded networking: %s", pod.Name, pod.Spec.NodeName, event.Message),
+			}
+			conditions = append(conditions, condition)
+			c.eventRecorder.Warningf(condition.Reason, condition.Message)
```

Comment:
this can come out multiple times because kubelets send multiple events.  You want to sum the events and *then* send these messages.

Reply:
@deads2k i thought the event recorder is smart enough on backend side to only increase the event counter. if this is broken, I don't mind creating more noise in the cluster?

Reply:
nvmd. I got what you meant... yeah this will just duplicate existing events and fire them using operator as component... i eliminated that and will only fire one warning event.